graphql/resolvers.go

```diff
@@ -357,6 +359,52 @@ func (r *queryResolver) TaskFiles(ctx context.Context, taskID string) ([]*Groupe
 	return groupedFilesList, nil
 }
 
+func (r *mutationResolver) SetTaskPriority(ctx context.Context, taskID string, priority int) (*restModel.APITask, error) {
+	t, err := r.sc.FindTaskById(taskID)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, err.Error())
+	}
+	if t == nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("cannot find task with id %s", taskID))
+	}
+	authUser := gimlet.GetUser(ctx)
+	if authUser == nil {
+		return nil, Forbidden.Send(ctx, "No user attached to request")
+	}
+	if priority > evergreen.MaxTaskPriority {
```

Comment:
I thought there was a function you could call to change the priority that did the security checks, but apparently there isn't. Would you be able to refactor this into its own function and call it from other places that are changing priority (search for "evergreen.TasksAdmin") or create a ticket to do this later?

Reply:
I found a function that does the checks but doesn't actually set the priority: https://github.com/evergreen-ci/evergreen/blob/master/rest/route/middleware.go#L120

Reply:
That helps, but there should be a single function to do all that (that should probably be combined into such a function too)

Reply:
I just made this ticket: https://jira.mongodb.org/browse/EVG-7448